lib/segment/src/index/field_index/full_text_index/inverted_index/mod.rs

```diff
@@ -333,6 +371,48 @@ pub trait InvertedIndex {
     fn get_token_id(&self, token: &str, hw_counter: &HardwareCounterCell) -> Option<TokenId>;
 }
 
+/// Intersect a collection of sorted iterators
+fn intersect_sorted<I, T>(mut iterators: Vec<I>) -> Vec<T>
```

Comment:
Could you elaborate why this was preferred over reusing the [intersection iterator](https://github.com/qdrant/qdrant/blob/9d35b84c7ed9a5d8d409649d0e18691f6cc4e374/lib/segment/src/index/field_index/full_text_index/inverted_index/postings_iterator.rs#L28)?

Reply:
@coszio I was able to leverage a lot more existing utilities through the `filter` function on the inverted index.
Please check again :)